### PR TITLE
Upgrade to Highlight.js 11.x and bump dependency versions

### PR DESCRIPTION
**Pull request by apjanke**
Upgrades to the current Highlight.js 11.x.

Upgrades several dependencies to their latest minor versions, which resolves security issues pointed out by `npm audit`.

With this change, `npm audit` reports no outstanding issues.

### CODE REVIEW

File: doc-project/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+This is a Node.js module, which is published on NPM. Any generic doco about Node.js modules and NPM publishing is relevant. [This article on auth0.com](https://auth0.com/blog/developing-npm-packages/) seems decent as an introductory guide. A lot of other articles you get when googling for "NPM module development" do not.
+
+The main developer (Andrew) works primarily on macOS, but this module and repo should work on all of Mac, Linux, and Windows.
+
+## Testing
+
+To test this language definition, link it into a highlight.js repo checkout and run its tests, as described in the [Highlight.js 3rd Party Quickstart Guide](https://github.com/highlightjs/highlight.js/blob/main/extra/3RD_PARTY_QUICK_START.md).
+
+These tests will currently fail, because this language definition will cause some examples to be detected as octave instead of matlab, breaking the matlag language definition tests. I can't yet figure out how to turn this off with `disableAutodetect: true` in the octave language definition. TODO: Figure this out.
```
Comment (joshgoebel) [inline doc-project/developer-guide.md:17]: I think this TODO is no longer relevant?

File: src/languages/octave.js [hunk 2]
```diff
@@ -23,6 +23,12 @@ export default function(hljs) {
 
   return {
     name: 'Octave',
+    /* 
+    Disable autodetection because it conflicts with matlab autodetection.
+    This breaks highlight.js tests, and Matlab should probably take precedence anyway because it's
```
Comment (joshgoebel) [inline src/languages/octave.js:28]: Also think this comment is outdated?

**Comment by joshgoebel** [comment]
Looks pretty good other than the two comments I dropped since I thought you had now figured out the auto-detect issue further.

**Comment by apjanke** [comment]
Sorry on the delay here; I am _utterly_ slammed at work this week. Should have some free time this weekend.

**Comment by joshgoebel** [comment]
No worries.

**Pull request closed without merging**